### PR TITLE
Integrate Minio for Enhanced File Storage Management

### LINKED ISSUE #259: Implementing Image upload/download and storage solution on top of the chat functionality
### Description

Enhancing the issue #239  and Related to issue #59 

- Implementation of image upload and download endpoints for the web-chat functionality
- Implementation of a storage solution for the images

### PR DESCRIPTION
**Pull request by parhamrahmani**
# Integrate Minio for Enhanced File Storage Management
## Description:

Closes #259 

This PR introduces Minio as the primary file storage solution within the remsfal-service application. 

#### File Storage Service
- ##### [Docker Compose Update](https://github.com/remsfal/remsfal-backend/pull/286/commits/bf7db6f6dd23d07996e1a4dc92dc96fd7aa03c87)
     - Added Minio Image: Incorporates the Minio container to facilitate file storage.

- ##### [File Storage Service](https://github.com/remsfal/remsfal-backend/pull/286/commits/a7c4a597b34df4c0dcd458b911bf5ecbfc651ae3)  
     - FileStorageService will handle all interactions with the Minio storage service. 
     - [Implemented FileStorageServiceTest to ensure all functionalities of the file storage service operate as expected, covering both success and failure scenarios.](https://github.com/remsfal/remsfal-backend/pull/286/commits/ec32eb88553003a4d0d462a1e7212d366ce912b3)

- ##### [Changes in Project Dependencies in remsfal-service](https://github.com/remsfal/remsfal-backend/pull/286/commits/91f90a54cbd85cca4aa2b4f8229fcb1bc1d64eff)
     - Quarkus-Minio Integration: Updated the pom.xml to include the quarkus-minio dependency, enabling Quarkus to interact effectively with the Minio service.

-  ##### [Configuration Updates](https://github.com/remsfal/remsfal-backend/pull/286/commits/53aa70a70f005f588f7db19e27f38998b252ac0d)
    - Application Properties: necessary Minio configurations

##### [Changes in CHAT_MESSAGE Table and Model](https://github.com/remsfal/remsfal-backend/pull/286/commits/ada843fc5123fe4a2a15d0f108bafe49e165e201)
- change CONTENT_TYPE enums to TEXT and FILE (instead of IMAGE)
- changing the column IMAGE_URL to URL
##### [Implementation of Upload Endpoint in ChatEndpoint](https://github.com/remsfal/remsfal-backend/pull/286/commits/5054077b4901e9777a0f795b28b99b2afbe9e6b5)
- upload a file with a request on `/{sessionId}/messages/upload`
- supported file types are determined in FileStorageService allowedTypes
- the file itself will be uploaded to the bucket ("remsfal-chat-files") and the metadata (file URL, relevant chat session ID, chat message ID, project and user ID, etc.) will be saved on CHAT_MESSAGE table.
##### [Downloading Files in a Chat Session](https://github.com/remsfal/remsfal-backend/pull/286/commits/f13f5f4b1cd24fc184a63aad383d087363cf2710)
- no new endpoint needed, on `/{sessionId}/messages/{messageId}` based on the messageId it will be determined from the CHAT_MESSAGE table if it's a text message or a file. If it's a file, a MediaType.APPLICATION_OCTET_STREAM will be produced or the file will be downloaded.

### CODE REVIEW

**Comment by parhamrahmani** [comment]
@astanik Hi, can you please review the new PR? I couldn't assign you myself. 

for some reason [this failure](https://github.com/remsfal/remsfal-backend/actions/runs/12357993407/job/34487492257?pr=286) happens which i didn't encounter in my local environment. 

This doesn't close #259 completely but it integrates minio into the project as a independent service. Then in the next step i will use this in the chat functionalites to upload and download files.

File: remsfal-service/src/test/java/de/remsfal/service/control/FileStorageServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+import jakarta.ws.rs.core.MultivaluedHashMap;
+import jakarta.ws.rs.core.MultivaluedMap;
+import java.util.Map;
+import org.jboss.logging.Logger;
+
+@QuarkusTest
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
```
Comment (astanik) [inline remsfal-service/src/test/java/de/remsfal/service/control/FileStorageServiceTest.java:26]: Why is this necessary?

**Comment by parhamrahmani** [comment]
@astanik Thank you for your response. I fixed the false configurations, still the problem persists. 

```
 Failed to initialize MinIO bucket: Failed to connect to localhost/[0:0:0:0:0:0:0:1]:9000
```
Since, I still don't encounter this issue in the local setup, I suspect the minio container is not set up in CI build or I still have made some misconfigurations.

**Comment by astanik** [comment]
@parhamrahmani As you may noticed, the testcontainer config is now correct.

**Comment by parhamrahmani** [comment]
@astanik Thank you for fixing the misconfigurations. This will close the issue #259 and also finishes the first prototype of "Fully-SQL" approach to the database solution for the chat service and also file storage for the chat service. The "Polyglot-Persistence" approach will be tested locally and if better will probably replace this approach, if not this will be the current solution. 

The PR is ready to review.

**Review by astanik — APPROVED** [review]
very good :thumbsup:
